OptimoveSDK/optimove-sdk/src/main/java/com/optimove/sdk/optimove_sdk/Optimove.java

```diff
@@ -121,48 +109,48 @@ public static Optimove getInstance() {
-                OptiLoggerStreamsContainer.debug("Optimove.configure() is starting");
-                shared.lifecycleObserver.addActivityStoppedListener(shared.optimoveLifecycleEventGenerator);
-                shared.lifecycleObserver.addActivityStartedListener(shared.optimoveLifecycleEventGenerator);
-                ((Application) applicationContext).registerActivityLifecycleCallbacks(shared.lifecycleObserver);
-                shared.fetchConfigs(false);
+        if (config.isOptimoveConfigured()){
+            OptiLoggerStreamsContainer.initializeLogger(application.getApplicationContext());
+
+            Runnable initCommand = () -> {
+                boolean initializedSuccessfully = performSingletonInitialization(application.getApplicationContext(),
+                        new TenantInfo(config.getOptimoveToken(), config.getConfigFileName()));
+                if (initializedSuccessfully) {
+                    OptiLoggerStreamsContainer.debug("Optimove.configure() is starting");
+                    shared.lifecycleObserver.addActivityStoppedListener(shared.optimoveLifecycleEventGenerator);
+                    shared.lifecycleObserver.addActivityStartedListener(shared.optimoveLifecycleEventGenerator);
+                    application.registerActivityLifecycleCallbacks(shared.lifecycleObserver);
+                    shared.fetchConfigs();
+                }
+            };
+            if (!OptiUtils.isRunningOnMainThread()) {
+                OptiLoggerStreamsContainer.debug("Optimove.configure() was called from a worker thread, moving call to main thread");
+                OptiUtils.runOnMainThread(initCommand);
+            } else {
+                initCommand.run();
             }
-        };
-        if (!OptiUtils.isRunningOnMainThread()) {
-            OptiLoggerStreamsContainer.debug("Optimove.configure() was called from a worker thread, moving call to main thread");
-            OptiUtils.runOnMainThread(initCommand);
-        } else {
-            initCommand.run();
         }
     }
 
     /**
      * Initializes the {@code Optimove SDK}. <b>Must</b> be called from the <b>Main</b> thread.<br>
      * Must be called as soon as possible ({@link Application#onCreate()} is the ideal place), and before any call to {@link Optimove#getInstance()}.
      *
-     * @param context           The instance of the current {@code Context} object.
-     * @param tenantInfo        The {@link TenantInfo} as provided by <i>Optimove</i>.
+     * @param application           The instance of the current {@code Application} object.
+     * @param optimobileConfig        The {@link OptimobileConfig} as provided by <i>Optimove</i>.
      * @param logcatMinLogLevel Logcat minimum log level to show.
      */
-    public static void configure(Context context, TenantInfo tenantInfo, LogLevel logcatMinLogLevel) {
+    public static void initialize(Application application, OptimobileConfig optimobileConfig, LogLevel logcatMinLogLevel) {
```

Comment:
Is it worth considering making log level part of the config object, saves having more overloads?

Reply:
suppose no reason not to add log level to config and have as a 'special' arg